### PR TITLE
feat(probe): support for ippool info

### LINKED ISSUE #231: Add support for used ips / port block allocations of a sourcenat rule
Hello,

Would it be possible to add support to show the amount of free port block allocations of an ip pool. 
This is availble at the following ap and retuns the following items: 

`/api/v2/monitor/firewall/ippool?vdom=*&access_token=<token>`

I am mainly interested in: 
`used`: amount of port blocks used
`total`: amount of port blocks available
`available`(OPTIONAL): percentage (but not really needed as it can be calculated from 100-used/total)
`natip_in_use`: amount of ips in the pba in use
`natip_total`: amount of ips available in the pba pool

```
[
    {
        "http_method": "GET",
        "results": {
            "PBA_NAME": {
                "name": "PBA_NAME",
                "blocks": 8,
                "block_size": 128,
                "fixed_port": false,
                "pba_per_ip": 472,
                "used": 10,
                "total": 118472,
                "available": 99.991561889648438,
                "clients": 10,
                "natip_in_use": 1,
                "natip_total": 251
            }
        },
        "vdom": "Hotspot",
        "path": "firewall",
        "name": "ippool",
        "action": "",
        "status": "success",
        "serial": "serialhere",
        "version": "v7.0.11",
        "build": 489
    }
]    
```

a 1:1 nat looks like this: 
```
            "PUB_NAME": {
                "name": "PUB_NAME",
                "blocks": 1,
                "block_size": 60416,
                "fixed_port": true,
                "pba_per_ip": 1,
                "used": 1,
                "total": 1,
                "available": 0.0,
                "clients": 1,
                "natip_in_use": 1,
                "natip_total": 1
            },
```

### PR DESCRIPTION
**Pull request by 33Fraise33**
This merge adds support for ippool information. This closes #231

### CODE REVIEW

File: pkg/probe/firewall_ippool.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	var (
+		mUsed = prometheus.NewDesc(
+			"fortigate_ippool_used",
+			"Amount of items used in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mTotal = prometheus.NewDesc(
+			"fortigate_ippool_total",
```
Comment (bluecmd) [inline pkg/probe/firewall_ippool.go:64]: Have a look at https://prometheus.io/docs/practices/naming/ - the `_total` suffix is used for counter values (not gauge values) that are strictly accumulating. I would propose something like `fortigate_ippool_total_ips` or `fortigate_ippool_total_count` or something like that to add the unit.

Comment (33Fraise33) [reply pkg/probe/firewall_ippool.go]: This should be fixed now

**Comment by 33Fraise33** [comment]
Added tests as well @bluecmd

File: pkg/probe/firewall_ippool.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	Results map[string]IpPool `json:"results"`
+	VDOM    string            `json:"vdom"`
+	Version string            `json:"version"`
+}
+
+func probeFirewallIpPool(c http.FortiHTTP, meta *TargetMetadata) ([]prometheus.Metric, bool) {
+	var (
+		mAvailable = prometheus.NewDesc(
+			"fortigate_ippool_available",
+			"Percentage available in ippool",
```
Comment (bluecmd) [inline pkg/probe/firewall_ippool.go:30]: `fortigate_ippool_available_percentage`

File: README.md [hunk 3]
```diff
@@ -63,6 +63,13 @@ Per-VDOM:
    * `fortigate_policy_bytes_total`
    * `fortigate_policy_hit_count_total`
    * `fortigate_policy_packets_total`
+ * _Firewall/IpPool_
+   * `fortigate_ippool_available`
```
Comment (bluecmd) [inline README.md:67]: These are not updated from the rename

File: pkg/probe/firewall_ippool.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	var (
+		mIpUsed = prometheus.NewDesc(
+			"fortigate_ippool_used_ip",
+			"Ip addresses in use in ippool",
+			[]string{"vdom", "name"}, nil,
+		)
+	)
+	var (
+		mIpTotal = prometheus.NewDesc(
+			"fortigate_ippool_total_ip",
```
Comment (bluecmd) [inline pkg/probe/firewall_ippool.go:43]: Nit pick: You use _clients (plural, which I like as well) and _ip (singular). I *personally* would use _ips but I will leave this decision on naming on you.

**Review by bluecmd — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (33Fraise33) [reply README.md]: done

File: pkg/probe/firewall_ippool.go (hunk 4 above)
Comment (33Fraise33) [reply pkg/probe/firewall_ippool.go]: changed to be plural everywhere

File: pkg/probe/firewall_ippool.go (hunk 2 above)
Comment (33Fraise33) [reply pkg/probe/firewall_ippool.go:30]: used `_ratio` to match with other metrics in the project

Comment (bluecmd) [reply pkg/probe/firewall_ippool.go:30]: That's fine, and I like that - but a ratio is defined as a value between 0.0 and 1.0, whereas percentage is 0 - 100. So you need to divide the value by 100.0 in this case (which I think is a good idea).

Comment (33Fraise33) [reply pkg/probe/firewall_ippool.go]: @bluecmd sorry it took me a while to finish this, but now this is 0 - 1.0

**Review by bluecmd — APPROVED** [review]

**Comment by bluecmd** [comment]
Merged, thanks!

**Pull request merged**